### PR TITLE
Python tests

### PR DESCRIPTION
**Pull request by fnand**
Several tests implemented and a few changed to fix issues found by tests.

### CODE REVIEW

**Comment by fnand** [comment]
Just double check if the final commit ( the one about  	"Add _limit definition to __init__") makes sense

File: python/histogrammar/primitives/absoluteerr.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def fill(self, datum, weight=1.0):
 
             # no possibility of exception from here on out (for rollback)
             self.entries += weight
-            self.absoluteSum += abs(q)
+            self.absoluteSum += abs(q) * weight
```
Comment (jpivarski) [inline python/histogrammar/primitives/absoluteerr.py]: Really, I missed that? Ouch! Good catch!

File: python/histogrammar/primitives/bag.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 # limitations under the License.
 
 import math
+import json
```
Comment (jpivarski) [inline python/histogrammar/primitives/bag.py]: "import json" is no longer needed, right?

Comment (fnand) [inline python/histogrammar/primitives/bag.py]: Yeah, that doesn't need to be there anymore

**Pull request merged**